### ISSUE TITLE
[NodeJS] templating for src/index.js

### ISSUE THREAD

**Issue by kinwa91** [GoogleCloudPlatform/synthtool#11]
Very often I have to manually edit `src/index.js` when new version of an API is being pulled in. I'd like that to be templated, however the `index.js` file for each API is different. Here is an example from Cloud Vision:

```javascript
/**
 * @type {object}
 * @property {constructor} ImageAnnotatorClient
 *   Reference to {@link v1.ImageAnnotatorClient}
 */
module.exports.v1 = gapic.v1;

/**
 * @type {object}
 * @property {constructor} ImageAnnotatorClient
 *   Reference to {@link v1p1beta1.ImageAnnotatorClient}
 */
module.exports.v1p1beta1 = gapic.v1p1beta1;

/**
 * @type {object}
 * @property {constructor} ImageAnnotatorClient
 *   Reference to {@link v1p2beta1.ImageAnnotatorClient}
 */
module.exports.v1p2beta1 = gapic.v1p2beta1;
```

Here we have a reference to ImageAnnotatorClient in the jsdoc section of the code, and it is generated based on gapic configs (I think?).

The templating strategy could be:
 - generate `src/index.js` using artman for a single version
 - for each version, take the module.exports... line that correspond to the version (regex?), along with its jsdoc section
 - bring it all into the generated `src/index.js`

**Comment by JustinBeckwith** [comment]
cc @crwilcox since we were just talking about this

**Comment by crwilcox** [comment]
Templating support is in. I made one for circle ci configs -> https://github.com/GoogleCloudPlatform/synthtool/blob/master/synthtool/gcp/templates/node_library/.circleci/config.yml.j2

we use jinja for templating. tags are like this ``{{ VARNAME }}``

If you look at https://github.com/googleapis/nodejs-speech/blob/1511cba881c5f6da0a711965503a6e599dcdb345/synth.py you can see how you can use template groups.

In short though

1) create index.js.j2 under nodejs folder in templates
2) Add the following code to synth.py:
```
common_templates = gcp.CommonTemplates()
templates = common_templates.node_library(package_name="@google-cloud/speech", my_arg = 'foo', other_arg = 'bar')
``` 
3) node_library takes keyword arguments. So just add new args by the names you need in your template.

**Comment by theacodes** [comment]
Closing in favor of #10
